Added linking to Python distributions according to the bit depth of W…

dynamite86 opened the pull request:
…indows.

dynamite86 committed:
Added linking to Python distributions according to the bit depth of Windows.

tz-lom commented on pythonEngine.qbs:
```diff
@@ -10,36 +10,84 @@ Product {
 
     cpp.cxxLanguageVersion: "c++11"
 
-    cpp.includePaths: [
```
You should not remove these paths, except the hard coded one
Use outer.concat to add paths, see linux build

tz-lom commented on pythonEngine.qbs:
```diff
@@ -10,36 +10,84 @@ Product {
-        '/usr/lib64/python2.7/site-packages/numpy/core/include'
-    ]
-    
+//    cpp.includePaths: [
```
No need to comment some code just in case, if you sure - remove it, if you are not sure - think again

tz-lom commented on pythonEngine-unittest.qbs:
```diff
@@ -4,27 +4,93 @@ import qbs.Probes as Probes
 CppApplication {
```
Currently this test make no sence, please revert changes and exclude it from the main qbs project, but don't remove this file

tz-lom commented on pythonEngine.qbs:
```diff
@@ -10,36 +10,84 @@ Product {
-        cpp.libraryPaths: pythonHome + "/libs"
+
+        //TODO: need to fix
+        //name: mobuleBaseName + "_x86"
```
why not to fix right now?

tz-lom commented on pythonEngine.qbs:
```diff
@@ -10,36 +10,84 @@ Product {
+            '_hypot=hypot'
+        ]
+
+        //TODO: need to fix
```
same as above

tz-lom commented on pythonEngine.qbs:
```diff
@@ -10,36 +10,84 @@ Product {
         Resonance.standalone: true
     }
 
+    //TODO: need to fix
```
What need to fix here?
Why this property is important?

dynamite86 committed:
1. Fixed indication of target paths (include, libs ect)
2. The path to the installation directory and the bitness of the Python distribution is determined automatically when building the project (debugging may be necessary).
3. The path to the installation directory of the NumPy package is determined automatically when building the project (debugging may be necessary).
4. Changes in the project "pythonEngine-unittest.qbs" were canceled before the original version (the project was excluded from the assembly in the project head file, the project file was not deleted).
5. Temporarily canceled the naming of the output file in accordance with the bit OS (it is necessary to clarify the appropriateness of this feature).

tz-lom commented on pythonEngine.qbs:
```diff
@@ -16,6 +17,57 @@ Product {
+        property string app: "python"
+
+        property string pythonRoot
+        property var getPythonRoot: ["-c", "import sys; print(sys.exec_prefix)"]
```
This property makes no sense as a property, move it to configure

tz-lom commented on pythonEngine.qbs:
```diff
@@ -16,6 +17,57 @@ Product {
+        property var getPythonRoot: ["-c", "import sys; print(sys.exec_prefix)"]
+
+        property string pythonArch
+        property var getPythonArch: ["-c", "import platform; print(platform.architecture()[0])"]
```
Same here

tz-lom commented on pythonEngine.qbs:
```diff
@@ -16,6 +17,57 @@ Product {
+        property var getPythonArch: ["-c", "import platform; print(platform.architecture()[0])"]
+
+        property string numpyIncludeDir
+        property var getNumpyIncludeDir: ["-c", "import numpy.distutils; print(numpy.distutils.misc_util.get_numpy_include_dirs())[0]"]
```
And here

tz-lom commented on pythonEngine.qbs:
```diff
@@ -16,6 +17,57 @@ Product {
+
+            if(proc.exec(app, getNumpyIncludeDir) === 0){
+                numpyIncludeDir = proc.readStdOut().trim();
+                found = true;
```
Well, imagine you failed to get python root but succeed to get numpy dir, why it should be true?

dynamite86 committed:
Fixed bugs in the algorithm for obtaining and verifying information about the installed Python distribution and the NumPy module.

tz-lom committed:
Merge branch 'master' into master
